examples/seismic/model.py

```diff
@@ -330,14 +334,11 @@ def __init__(self, origin, spacing, shape, space_order, vp, nbpml=20,
         self.grid = Grid(extent=extent, shape=shape_pml,
                          origin=origin, dtype=dtype)
 
-        # Create square slowness of the wave as symbol `m`
-        if isinstance(vp, np.ndarray):
+        if vp is None:
             self.m = Function(name="m", grid=self.grid, space_order=self.space_order)
+            self.m.data[:] = m
```

Comment:
`initialize_function(self.m, m, self.nbpml)`

Reply:
That fucks things up because it adds an additional nbpml padding that I don't want.

Reply:
This is at the level of generic model initialization. I f ones decide to provide `m` instead of `vp`, this has to be done. 
This is also inconsitent with the `vp` otpion that does add the abc layer.